<html xmlns="http://www.w3.org/1999/xhtml"><head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
    <title>OPERA</title>
    <link type="text/css" rel="stylesheet" href="./static/css/jquery.tocify.css">
    <link type="text/css" rel="stylesheet" href="./static/css/bootstrap.min.css">
    <link type="text/css" rel="stylesheet" href="./static/css/solarized-light.css">
    <link type="text/css" rel="stylesheet" href="./static/css/main.css">
    <script src="./static/js/jquery-1.12.4.min.js"></script>
    <script src="./static/js/jquery-ui.min.js"></script>
    <script src="./static/js/jquery.tocify.js"></script>
    <script src="./static/js/hljs.js"></script>
    <script type="text/x-mathjax-config">
        MathJax.Hub.Config({tex2jax: {inlineMath: [['\\$','\\$'], ['\\(','\\)']]}});
      </script>
    <script type="text/javascript" async="" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML">
    </script>
</head>
<body>
    <div class="page-header" id="PAGE-header" style="height:auto;padding:0">
        <div class="container-fluid">
            <p class="TOP_head">&nbsp;</p>
            <div class="span5">
                <h1>&nbsp;OPERA</h1>
                <p class="TOP_head">Omics PlEiotRopic Association study</p>
            </div>
            <div class="span7">
                <ul class="nav nav-pills" style="margin:0">
                    <li role="presentation"><a href="https://yanglab.westlake.edu.cn/software/smr/#Multi-SNP-basedSMRtest">SMR-multi</a></li>
                    <li role="presentation" class="active"><a href="https://github.com/wuyangf7/OPERA">OPERA</a></li>
                    <li role="presentation" class="active"><a href="https://gctbhub.cloud.edu.au/software/gctb/#Genome-wideFine-mappinganalysis">GWFM</a></li>
                </ul>
            </div>
        </div>
    </div>
    <div class="container-fluid">
        <div class="row-fluid">
            <div class="span3">
                <div id="toc"></div>
                <!--[if lt IE 9]>
				<div style=" margin-top: 230px; position: fixed; width: 230px; background-color: yellow; z-index: 10000;">
				<p>Due to our shortage of testing, the document can't view correctly under Internet Explorer <= 8.0. You may need update your browser, or downlad offline <a href="./static/gcta_doc_latest.pdf">PDF document</a>, <a href="./static/gcta_latest.zip">executable Files</a></p>
				</div>
				<![endif]-->
            </div>
            <div id="show" class="span9">
                <p> Loading...</p>
                <p> If it not work properly, you may need update your Internet browser and enable javascript</p>
                <p> or downlad offline <a href="./download/GCTB.pdff">PDF document</a>, <a href="./download/gctb_1.0_Linux.zip">executable Files</a></p>
            </div>
            <div id="content" class="span9" style="display: none;"><body><h2 id="overviewexpand" class="expand">Overview </h2>
<hr><h3 id="about">About</h3>
<p>OPERA is a Bayesian extension of the SMR &amp; HEIDI approach to a multi-omics model, where the molecular phenotypes are considered as exposures and the complex trait is considered as the outcome. This tool can therefore be used to prioritize molecular phenotypes that mediate the genetic effects for complex trait and further provide mechanistic interpretation of the GWAS signal.</p>
<hr><h3 id="credits">Credits</h3>
<ul>
<li>Yang Wu developed the software tool with the support from Ting Qi, Jian Zeng, and Jian Yang.</li>
<li>Yang Wu, Jian Zeng and Jian Yang developed the OPERA method. </li>
<li>The software is programmed and maintained by Yang Wu.</li>
</ul>
<hr><h3 id="questionsandhelprequests">Questions and Help Requests</h3>
<p>If you have any bug reports or questions, please send an email to Yang Wu (<a href="mailto:yang.wu@scu.edu.cn">yang.wu@scu.edu.cn</a>) at West China Hospital, Sichuan University  and Jian Zeng (<a href="mailto:j.zeng@uq.edu.au">j.zeng@uq.edu.au</a>) at Institute for Molecular Bioscience, The University of Queensland, and Jian Yang (<a href="mailto:jian.yang@westlake.edu.cn">jian.yang@westlake.edu.cn</a>) at School of Life Sciences, Westlake University.</p>
<hr><h3 id="citations">Citations</h3>
<p>Wu Y., Qi T., Wray N.R., Visscher P.M., Zeng J. &amp; Yang J. (2023) Joint analysis of GWAS and multi-omics QTL summary statistics reveals a large fraction of GWAS signals shared with molecular phenotypes. Cell Genomics.</p>
<h2 id="downloadexpand" class="expand">Download </h2>
<hr><h3 id="software">Software</h3>
<p>The binary executable files will be available soon.</p>
<hr><h3 id="dataresource">Data Resource</h3>
<p>We have curated and prepared a variety of publicly available molecular QTL data, which users can use to perform the OPERA analysis with their specific complex trait of interest.</p>
<ul>
<li><a href="https://cnsgenomics.com/software/smr/#DataResource"><strong>Download molecular QTL data</strong></a></li>
</ul>
<p>For illustration purposes, we provide demonstration data that can be used to run the tutorial analysis.</p>
<ul>
<li><a href="https://github.com/wuyangf7/OPERA/tree/main/demo"><strong>Download Demo Data</strong></a></li>
</ul>
<h2 id="installationexpand" class="expand">Installation </h2>
<p>To install OPERA, you can download the <a href="https://github.com/wuyangf7/OPERA/blob/main/opera_Linux.zip">opera<em>Linux.zip</em></a><em> package, which contains a standalone (i.e., statically linked) 64-bit Linux executable file opera</em>Linux. We strongly recommend using this static executable because it is well-optimized and no further installation is required.</p>
<p>For compiling yourself, you should clone this repository and use the MAKEFILE,  </p>
<pre><code>git clone https://github.com/wuyangf7/OPERA
cd OPERA
make
</code></pre>
<p>There are dependencies on your local <code>MKL</code>, <code>BOOST</code> and <code>EIGEN</code> Libraries.</p>
<h2 id="basicoptionsexpand" class="expand">Basic Options </h2>
<p>Below is a list of common flags used in OPERA analysis.</p>
<hr><h3 id="inputoutput">Input/Output</h3>
<ul>
<li><code>--besd-flist &lt;file&gt;</code>: Reads a file to get the full paths of the multiple xQTL BESD files.</li>
<li><code>--gwas-summary &lt;file&gt;</code>: Reads summary-level data from GWAS.</li>
<li><code>--bfile &lt;prefix&gt;</code>: Reads individual-level SNP genotype data (PLINK binary format) as LD reference.</li>
<li><code>--out &lt;prefix&gt;</code>: Saves the results with the specified prefix.</li>
</ul>
<hr><h3 id="analysiscontrol">Analysis Control</h3>
<ul>
<li><code>--thread-num &lt;int&gt;</code>: Specifies the number of OpenMP threads for parallel computing.</li>
<li><code>--sample-overlap</code>: Specifies the flag to let OPERA consider the between-study correlations due to overlapping samples.</li>
<li><code>--opera-smr</code>: Turns on the flag of using the estimated SMR effect rather than the estimated joint SMR effect.</li>
<li><code>--heidi-off</code>: Turns off the heterogeneity test.</li>
</ul>
<hr><h3 id="thresholds">Thresholds</h3>
<ul>
<li><code>--thresh-ppa &lt;float&gt;</code>: Significance threshold of PPA (default: 0.9).</li>
<li><code>--thresh-smr &lt;float&gt;</code>: Significance threshold of SMR (default: 0.05).</li>
<li><code>--thresh-heidi &lt;float&gt;</code>: Significance threshold of HEIDI test (default: 0.01).</li>
</ul>
<h2 id="tutorialexpand" class="expand">Tutorial </h2>
<p>The OPERA analysis consists of three steps. OPERA first estimates the global frequencies of each possible association pattern. It then caculates the posterior probability for each configuration by weighting the data likelihood with the estimated global frequencies. The posterior probability of associations (PPA) for any combinatorial sites is computed by summing up the posterior probability of configurations where the given site combination is present. For associations that passed a PPA signficance threshold (e.g., 0.9 by default), OPERA performs the heterogeneity test to reject the associations that are due to linkage. </p>
<p>We have curated and prepared a variety of publicly available molecular QTL data (<a href="https://cnsgenomics.com/software/smr/#DataResource">download here</a>), which users can use to perform the OPERA analysis with their specific complex trait of interest. For illustration purpose, we provide the demonstration <a href="https://github.com/wuyangf7/OPERA/tree/main/demo">data</a> that can be used to run opera analysis with command line below. </p>
<hr><h3 id="opera">OPERA</h3>
<h4 id="operaanalysisstage1">OPERA analysis: stage 1</h4>
<pre><code>opera --besd-flist mylist --gwas-summary mygwas.ma --mbfile mybdatalist --estimate-pi --out myopera
</code></pre>
<p><strong><span class="kwd">--besd-flist</span></strong> reads a file to get the full paths of the multiple xQTL BESD files. The input format follows that for the <a href="https://cnsgenomics.com/software/smr/#DataManagement">SMR analysis</a>.</p>
<p><strong><span class="kwd">--gwas-summary</span></strong> reads summary-level data from GWAS. The input format follows that for <a href="http://cnsgenomics.com/software/gcta/#COJO">GCTA-COJO analysis</a>.</p>
<p><strong><span class="kwd">--mbfile</span></strong> reads a text file with each row representing a PLINK binary file (e.g., one for each chromosome) from a reference sample for LD estimation, i.e. .bed, .bim, and .fam files. The chromosome-wide .fam files are required to contain the same individuals. The input file format is blow. Note that stage 1 analysis requires the genome-wide LD reference to estimate the global parameters. If the genome-wide genotype data in PLINK format (containing the genotype data for each chromosome) is ready, please switch the flag <strong><span class="kwd">--bfile</span></strong> to read the standard PLINK binary file as LD reference.</p>
<pre><code>mydata
mydata2
</code></pre>
<p><strong><span class="kwd">--out</span></strong> saves the estimation of prior probabilities from the OPERA stage 1 analysis in <code>.pi</code> file and the estimation of prior variances in <code>.var</code> file (text format, see example from demo data below).</p>
<pre><code>Posteriors      Pi1(0:0)        Pi2(0:1)        Pi3(1:0)        Pi4(1:1)
Mean    0.605441        0.168855        0.0743364       0.151368
SD      0.20765 0.186273        0.134037        0.16549
</code></pre>
<pre><code>Variance        2.000000e-02    2.000000e-02
</code></pre>
<p>The <code>.pi</code> output includes the posterior mean and SD from MCMC for each possible configuration. The <code>.var</code> output includes the prior variance for each xQTL data. The posterior samples from MCMC are also printed in the log file, for example, </p>
<pre><code>Iteration       Pi1(0:0)        Pi2(0:1)        Pi3(1:0)        Pi4(1:1)
0       0.0557406       0.319719        0.0834392       0.541101
100     0.163492        0.422161        0.413673        0.000673509
200     0.661116        0.338883        2.4066e-07      6.7008e-15
300     0.587679        0.412321        5.49276e-11     3.1432e-09
400     0.814359        0.101207        0.0586814       0.0257529
500     0.430485        8.37885e-05     0.0929275       0.476504
600     0.282069        0.657061        0.017055        0.0438149
...
</code></pre>
<p>Columns are iteration numbers and posterior samples for each configuration from the MCMC.  </p>
<h4 id="otherparametersforstage1analysis">Other parameters for stage 1 analysis</h4>
<pre><code>opera --besd-flist mylist --gwas-summary mygwas.ma --bfile mydata --sample-overlap --estimate-pi --extract-snp mysnplist --prior-var 0.02,0.02 --pi-wind 100 --out myopera --thread-num 3
</code></pre>
<p><strong><span class="kwd">--bfile</span></strong> reads individual-level SNP genotype data (in PLINK binary format) from a reference sample for LD estimation, i.e. .bed, .bim, and .fam files. </p>
<p><strong><span class="kwd">--extract-snp</span></strong> specifies a snplist (e.g., <a href="https://github.com/wuyangf7/OPERA/blob/main/demo/hapmap.snplist">Hapmap3 SNP list</a>) to be extracted across LD reference, xQTL data and GWAS summary data, and used for stage 1 analysis. </p>
<p><strong>\â€“-prior-var</strong> specifies the prior variance of the non-zero mediated effects for each molecular trait on the complex trait. It can be computed by the variance of the estimated SMR effects of each molecular trait on complex trait at a significance level (e.g., FDR &lt; 0.05) adjusting for the estimation errors.</p>
<p><strong><span class="kwd">--sample-overlap</span></strong> specifies the flag to let OPERA consider the between-study correlations due to overlapping samples. OPERA will automatically output the estimated correlations in <code>.rho</code> file (text format, see below example). </p>
<pre><code>1.000000e+00    0.200000e+00
0.200000e+00    1.000000e+00
</code></pre>
<p><strong><span class="kwd">--opera-smr</span></strong> turns on the flag of using the estimated smr effect rather than the estimated joint smr effect to run the stage 1 analysis.  </p>
<p><strong><span class="kwd">--pi-wind</span></strong> defines a window centered on the molecular phenotype with smallest number of sites to select no overlap independent loci, e.g., 200 Kb (default).</p>
<p><strong><span class="kwd">--thread-num</span></strong> specifies the number of OpenMP threads for parallel computing.</p>
<h4 id="operaanalysisstagte2andheterogeneityanalysis">OPERA analysis: stagte 2 and heterogeneity analysis</h4>
<pre><code>opera --besd-flist mylist --extract-gwas-loci myloci --chr 7 --gwas-summary mygwas.ma --bfile mydata --prior-pi-file myopera.pi --prior-var-file myopera.var --out myopera_chr7
</code></pre>
<blockquote>
  <p>Note: Only the <code>cis-SNPs</code> of each exposure site are used, so the stage 2 analysis can be performed for each <code>chromosome seperately</code>. The genome-wide analysis results can be combined through shell command below, e.g., for association between single exposure and outcome,</p>
</blockquote>
<pre><code>awk 'NR==1 || FNR&gt;1' myopera_chr*_1_expos_assoc.ppa &gt; myopera_1_expos_assoc.ppa  
</code></pre>
<p><strong><span class="kwd">--extract-gwas-loci</span></strong> extracts a subset of genomic region with GWAS COJO or LD clumpped loci (i.e., 2Mb region by default) to perform opera analysis. The input file format is</p>
<pre><code>Chr     SNP     bp
7       rs1859788       99971834
7       rs7810606       143108158
</code></pre>
<p><strong><span class="kwd">--chr</span></strong> specifies the target chromosome for SNPs and exposure sites to run chromosome-wide opera stage 2 analysis.</p>
<p><strong><span class="kwd">--bfile</span></strong> reads individual-level SNP genotype data (in <code>PLINK</code> binary format) from a reference sample for LD estimation. </p>
<p><strong><span class="kwd">--prior-pi-file</span></strong> reads the prior probabilities estimated from the stage 1 analysis (i.e., the posterior Mean from stage 1 analysis). </p>
<p><strong><span class="kwd">--prior-var-file</span></strong> reads the prior variances estimated from the stage 1 analysis.  </p>
<p><strong><span class="kwd">--out</span></strong> saves the marginal or joint PPA and multi-exposure HEIDI test P-values for each association hypothesis passed significance threshold (e.g., <code>PPA &gt; 0.9 and P-HEIDI &gt; 0.01</code> default) in <code>.ppa</code> file (text format, see below example). Results are seperated as different number of exposure combinations associated with the outcome, e.g., single exposure association, </p>
<pre><code>Chr    GWAS_SNP    GWAS_bp Expo1_ID    Expo1_bp    PPA(1)  p_HEIDI(1)
7    rs1859788   99971834    cg08582801  99588335    0.929121    8.152293e-02
7    rs1859788   99971834    cg13210467  99775443    0.999054    1.625980e-01
7    rs1859788   99971834    cg19116668  99932089    1   1.125248e-01
7    rs1859788   99971834    cg26429636  99573747    0.915179    5.926392e-02
...
</code></pre>
<p>and associations between 2 exposure(s) and 1 outcome.</p>
<pre><code>Chr    GWAS_SNP    GWAS_bp Expo1_ID    Expo1_bp    Expo2_ID    Expo2_bp    PPA(1,2)    p_HEIDI(1,2)
7    rs1859788   99971834    ENSG00000085514 99981436    cg13210467  99775443    0.998746    1.787164e-02
7    rs1859788   99971834    ENSG00000146828 100444536   cg01869186  100423987   0.906761    1.641383e-01
7    rs1859788   99971834    ENSG00000146828 100444536   cg12616177  100434510   0.902426    1.044594e-01
...
</code></pre>
<p>Columns are chromosome, rsID for GWAS SNP, physical position for GWAS SNP, probe ID for the 1st exposure, probe position for the 1st exposure, probe ID for the 2nd exposure, probe position for the 2nd exposure, PPA for the 1st and 2nd exposures joint association, and p-value from HEIDI for the 1st and 2nd exposures joint association.  </p>
<blockquote>
  <p>Note: If gwas loci file is not specified, the program will scan all the possible combinations between exposure sites for the whole genome. </p>
</blockquote>
<p>The program will automatically ouput the proportion and number of GWAS loci associated with different combination of xQTL data in <code>.prop</code> file (see example below).    </p>
<pre><code>Overall    Exposures(1)    Exposures(2)    Exposures(1,2)
0.5    0   0.5 0.5
1    0   1   1
</code></pre>
<p>The output also includes the estimated false discovery rate (FDR) and false positive rate (FPR) for any combinatorial associations, which are printed in the log file, for example,</p>
<pre><code>PPA results for 4 combinatorial associations between 1 exposure(s) and 1 outcome have been extracted and saved in the file /Users/uqywu16/Desktop/mtSMR/software/demo/myopera_1_expos_assoc.ppa.
The estimated FDR is 0.0391616 for combinatorial associations between 1 exposure(s) and 1 outcome.
The estimated FPR is 0.0168247 for combinatorial associations between 1 exposure(s) and 1 outcome.

PPA results for 3 combinatorial associations between 2 exposure(s) and 1 outcome have been extracted and saved in the file /Users/uqywu16/Desktop/mtSMR/software/demo/myopera_2_expos_assoc.ppa.
The estimated FDR is 0.0640222 for combinatorial associations between 2 exposure(s) and 1 outcome.
The estimated FPR is 0.00503074 for combinatorial associations between 2 exposure(s) and 1 outcome.

There are 50% GWAS loci were detected to be associated with at least one type of xQTL data.
</code></pre>
<blockquote>
  <p>Note: we suggest a PPA threshold of 0.9 to roughly control the FDR below 0.05. However, if more strigent FDR or FPR is required and interested, OPERA can acheive this by increasing the PPA threshold (e.g., 0.995). The heterogeneity test (i.e., multi-exposure HEIDI) will be automatically performed for any combinatorial associations passed a PPA threshold (0.9 as default). If the heterogeneity test is not interested, it can be turned off by specifying <strong><span class="kwd">--heidi-off</span></strong>.</p>
</blockquote>
<h4 id="otherparametersforstage2analysis">Other parameters for stage 2 analysis</h4>
<pre><code>opera --besd-flist mylist --gwas-summary mygwas.ma --bfile mydata --chr 7 --extract-exposure-probe myexposure --sample-overlap --rho-file myopera.rho --outcome-wind 1000 --thresh-ppa 0.5 --thresh-smr 0.05 --thresh-heidi 0.01 --print-combo-ppa-res --extract-target-cojo-snps mycojo --extract-gwas-loci myloci --prior-pi 0.8,0.09,0.09,0.02 --prior-var 0.02,0.02 --out myopera --thread-num 3
</code></pre>
<p><strong><span class="kwd">--prior-pi</span></strong> specifies the prior probalities for stage 2 analysis.</p>
<p><strong><span class="kwd">--prior-var</span></strong> specifies the prior variance for stage 2 analysis (seperated by comma).</p>
<p><strong><span class="kwd">--rho-file</span></strong> specifies the estimated between-study correlations due to sample overlap from the stage 1 analysis. See the input format for the .rho file above.</p>
<p><strong><span class="kwd">--extract-exposure-probe</span></strong>    extracts a subset of exposure sites for opera analysis.</p>
<p><strong><span class="kwd">--outcome-wind</span></strong> specifies the window around each GWAS loci for stage 2 analysis, or the window around each site across exposures for stage 2 analysis when GWAS loci were not specified, e.g., 1Mb in either direction (default). </p>
<p><strong><span class="kwd">--extract-target-cojo-snps</span></strong> specifies full COJO SNP list for each site of molecular phenotype as the conditional SNP set to compute the joint SMR effect for other exposure sites. The input file format is</p>
<pre><code>ENSG00000196367 rs182325057,rs12532598,rs17638906,rs62472014,rs219843,rs183732601,rs150746244
ENSG00000238109 rs2283015,rs142345619,rs117749026,rs117295696,rs139511767
ENSG00000242687 rs34631688,rs187375676,rs149211972,rs219813,rs6976207,rs7789895,rs10264067,rs150746244
</code></pre>
<p><strong><span class="kwd">--thresh-ppa</span></strong> specifies significance threshold of PPA to perform heterogeneity test and output the significant results. </p>
<p><strong><span class="kwd">--thresh-smr</span></strong> specifies significance threshold of SMR to perform the OPERA analysis, e.g., 0.05 (default).</p>
<p><strong><span class="kwd">--thresh-heidi</span></strong> specifies significance threshold of single-exposure HEIDI test to perform the OPERA analysis, e.g., 0.01 (default).</p>
<p><strong><span class="kwd">--opera-smr</span></strong> turns on the flag of runing OPERA analysis using the estimated SMR effect rather than estimated joint-SMR effect. </p>
<p><strong><span class="kwd">--thread-num</span></strong> specifies the number of OpenMP threads for parallel computing. </p>
<p><strong><span class="kwd">--print-smr-res</span></strong> saves the pairwise smr result for all tested exposure sites in .smr file if specified. </p>
<pre><code>probeID    ProbeChr    Gene    Probe_bp    topSNP  topSNP_chr  topSNP_bp   A1  A2  Freq    b_GWAS  se_GWAS p_GWAS  b_eQTL  se_eQTL p_eQTL  b_SMR   se_SMR  p_SMR   p_HEIDI nsnp_HEIDI
ENSG00000238109    7   AC004893.10 98596857    rs2283015   7   98600839    G   C   0.0847203   -0.00193708 0.00365864  5.964895e-01    0.711699    0.0382071   1.926562e-77    -0.00272178 0.00514278  5.966378e-01    7.575220e-01    20
ENSG00000146833    7   TRIM4   99495902    rs2571997   7   99514417    A   C   0.410541    0.00614763  0.00213414  3.969059e-03    -0.690266   0.00715596  0.000000e+00    -0.00890618 0.00309315  3.985252e-03    8.417932e-05    20
ENSG00000166526    7   ZNF3    99670913    rs67110214  7   99633739    C   G   0.28583 0.00401872  0.0024395   9.948536e-02    -0.0968669  0.0088226   4.801025e-28    -0.041487   0.0254659   1.032880e-01    4.811370e-05    20
cg03856969    7   AK001533    98600799    rs4729505   7   98601025    C   T   0.0844313   -0.0024541  0.00373076  5.106651e-01    -0.431667   0.0598434   5.462031e-13    0.00568517  0.00867854  5.124136e-01    6.233012e-01    20
cg10154880    7   AK001533    98603502    rs17720576  7   98616657    A   G   0.0865118   -0.0018608  0.00371972  6.168967e-01    0.489588    0.05912 1.218873e-16    -0.00380076 0.00761151  6.175377e-01    8.987915e-01    20
...
</code></pre>
<p>Columns are probe ID, probe chromosome, gene name, probe position, SNP name, SNP chromosome, SNP position, the effect (coded) allele, the other allele, frequency of the effect allele (estimated from the reference samples), effect size from GWAS, SE from GWAS, p-value from GWAS, effect size from eQTL study, SE from eQTL study, p-value from eQTL study, effect size from SMR, SE from SMR, p-value from SMR, p-value from HEIDI test, and number of SNPs used in the HEIDI test.</p>
<p><strong><span class="kwd">--print-combo-ppa-res</span></strong> saves the ppa for each possible association hypothesis across all combinations in <code>.res</code> file if specified. </p>
<pre><code>Chr    GWAS_SNP    GWAS_bp Expo1_ID    Expo1_bp    Expo2_ID    Expo2_bp    PPA(0)  PPA(1)  PPA(2)  PPA(1,2)    p_HEIDI(1)  p_HEIDI(2)  p_HEIDI(1,2)
7    rs1859788   99971834    ENSG00000085514 99981436    cg27486786  99074036    0.4934397   0.37980783  0.42463091  0.2978785   NA  NA  NA
7    rs1859788   99971834    ENSG00000085514 99981436    cg05832509  99515080    0.5101769   0.36135745  0.41064838  0.28218272  NA  NA  NA
7    rs1859788   99971834    ENSG00000085514 99981436    cg26429636  99573747    0.14851999  0.56356138  0.83166325  0.54374462  NA  NA  NA
7    rs1859788   99971834    ENSG00000085514 99981436    cg08582801  99588335    0.13705845  0.57444245  0.84428662  0.55578756  NA  NA  NA
7    rs1859788   99971834    ENSG00000085514 99981436    cg13210467  99775443    2.3377886e-06   0.99889582  0.99984795  0.9987461   5.470704e-02    1.625980e-01    1.787164e-02
7    rs1859788   99971834    ENSG00000085514 99981436    cg19116668  99932089    1.4435816e-11   0.54813665  1   0.54813665  NA  1.125248e-01    NA
...
</code></pre>
<p>Columns are chromosome, rs ID for GWAS SNP, physical position for GWAS SNP, probe ID for the 1st exposure, probe position for the 1st exposure, probe ID for the 2nd exposure, probe position for the 2nd exposure, PPA for no associations, PPA for the 1st exposure marginal association, PPA for the 2nd exposure marginal association, PPA for the 1st and 2nd exposures joint association, p-value from HEIDI for the 1st exposure association, p-value from HEIDI for the 2nd exposure association, and p-value from HEIDI for the 1st and 2nd exposures joint association. Missing value is represented by "NA". </p>
<hr><h3 id="omicslocusplot">Omics locus plot</h3>
<h4 id="commandlinetogenerateadatafileforxqtlplot">Command line to generate a data file for xQTL plot</h4>
<pre><code>opera --besd-flist mylist1 --gwas-summary mygwas.ma --bfile mydata --beqtl-summary xQTL1 --out myplot --plot --probe ENSG00000085514 --probe-wind 500 --gene-list glist-hg19
</code></pre>
<p><strong><span class="kwd">--beqtl-summary</span></strong> reads the BESD file for one type of xQTL (e.g.,eQTL).</p>
<p><strong><span class="kwd">--besd-flist</span></strong> reads a file to get the full paths of the other xQTL BESD files.</p>
<p><strong><span class="kwd">--probe</span></strong> specifies a probe site (e.g., often a gene) as locus center to generate the plot.</p>
<p><strong><span class="kwd">--probe-wind</span></strong> specifies a window to extract the exposure site in the window</p>
<p><strong><span class="kwd">--gene-list</span></strong> specifies a gene range list, which is available for download <a href="https://github.com/wuyangf7/OPERA/blob/main/demo/glist-hg19">here</a>.</p>
<p>To generate the loucs omics plot with extracted file, please see the <a href="https://yanglab.westlake.edu.cn/software/smr/#OmicsSMRplot">omics SMR plot page</a>.</p>
<p>We also provide an <a href="https://github.com/wuyangf7/OPERA/blob/main/demo/plot/plot_OmicsSMR_xQTL.r">R scirpt</a> to plot the omics SMR plot as presented in Wu et al.. Please see the demo plot below.  </p>
<h4 id="rcommandstodrawtheplots">R commands to draw the plots</h4>
<pre><code>source("./plot/plot_OmicsSMR_xQTL.r") 
SMRData = ReadomicSMRData("./plot/myplot.ENSG00000085514.txt")
omicSMRLocusPlot(data=SMRData,esmr_thresh=1e-4,msmr_thresh=1e-4,eprobeNEARBY="ENSG00000085514",mprobeNEARBY=c("cg13210467"),trait_name="Test",funcAnnoFile="./plot/funcAnno.RData")
</code></pre>
<p><strong>esmr_thresh</strong> and msmr_thresh are the threshold for first xQTL and other xQTLs, respectively</p>
<p><strong>eprobeNEARBY</strong> specifies the eQTL association pattern for specific gene</p>
<p><strong>mprobeNEARBY</strong> specifies the xQTL association pattern for other specific sites</p>
<p><strong>funcAnnoFile</strong> reads the epigenome annotation data from the Roadmap Epigenomics Consortium, which can be downloaded <a href="https://www.dropbox.com/s/0rf4hmdjzostuv2/funcAnno.RData?dl=0">here</a>.</p>
<p><strong>OPERA</strong> shares the same data management function and flags with the <strong>SMR</strong> software, for a full list of option reference, please see <a href="https://cnsgenomics.com/software/smr/#OptionsReference">here</a>. </p></body></div>
        </div>

        <script>
            $(function() {
                var toc = $("#toc").tocify({
                    selectors: "h2, h3",
                    ignoreSelector: ".notoc",
                    expand: ".expand",
                    showEffect: "fadeIn",
                    scrollHistory: false,
                    extendPage: false,
                    scrollRef: "#PAGE-header",
                    showDIV: "#show",
                    tocID: "#toc"
                }).data("toc-tocify");
                toc.setOptions({
                    appendContents: '<div style="text-align: center;font-size: 14px"></div>'
                });
                hljs.initHighlightingOnLoad();

            });

            $(function() {
                setTimeout(function() {
                    (function(i, s, o, g, r, a, m) {
                        i['GoogleAnalyticsObject'] = r;
                        i[r] = i[r] || function() {
                            (i[r].q = i[r].q || []).push(arguments)
                        }, i[r].l = 1 * new Date();
                        a = s.createElement(o),
                            m = s.getElementsByTagName(o)[0];
                        a.async = 1;
                        a.src = g;
                        m.parentNode.insertBefore(a, m)
                    })(window, document, 'script', 'https://www.google-analytics.com/analytics.js', 'ga');
                    ga('create', 'UA-101733990-7', 'auto');
                    ga('send', 'pageview');
                }, 2000);
            });
        </script>
    </div>




</body></html>